### PR TITLE
[Java] Avoid memory allocation at varDataDisplay

### PR DESCRIPTION
**Pull request by ariksher-nex**
varDataDisplay - use method `getVarFieldName(Appendable)` - no memory allocation. unlike calling method `varFieldName` where it allocates memory (new byte[] and new String).

### CODE REVIEW

**Comment by mjpt777** [comment]
This generates code which does not compile.

**Comment by ariksher-nex** [comment]
> This generates code which does not compile.

yep, fixing it. apperantly code differs for UTF and ASCII.

**Comment by mjpt777** [comment]
The UTF-8 implementation allocates too. I'm not in favour of supporting this as it is too brittle.

**Comment by ariksher-nex** [comment]
> The UTF-8 implementation allocates too. I'm not in favour of supporting this as it is too brittle.

I'm using at my project ASCII chars and I'm willing to use the method **appendTo** that sits on the generated xxxDecoder class(es), which is very useful for logging incoming messages.

**Comment by mjpt777** [comment]
I've pushed a more obvious changes which saves on allocation.

**Comment by ariksher-nex** [comment]
> I've pushed a more obvious changes which saves on allocation.

seen your commit and it paratially solves the problem see my comment there
https://github.com/real-logic/simple-binary-encoding/commit/391527ac04df032671167b12d923b29dad30c5c0#r35648950

**Comment by mjpt777** [comment]
A better approach for you to take is to use the append version if and only iff the encoding is ASCII otherwise output as is at present.

File: sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java [hunk 1]
```diff
@@ -904,6 +904,28 @@ private void generateDataDecodeMethods(
                     generateGet(lengthType, "limit", byteOrderStr),
                     byteOrderStr);
             }
+            if (characterEncoding.contains("UTF"))
```
Comment (mjpt777) [inline sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java]: Why is this method still present?

Comment (ariksher-nex) [reply sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java]: removed it.

File: sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java [hunk 2]
```diff
@@ -3269,7 +3269,14 @@ private void appendDecoderDisplay(
             }
             else
             {
-                append(sb, indent, "builder.append('\\'').append(" + varDataName + "()).append('\\'');");
+                if (characterEncoding.contains("ASCII"))
+                {
+                    append(sb, indent, "builder.append('\\'');");
+                    append(sb, indent, formatGetterName(varDataToken.name()) + "(builder);");
+                    append(sb, indent, "builder.append('\\'');");
+                } else {
```
Comment (mjpt777) [inline sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java]: This will fail formatting.

Comment (ariksher-nex) [reply sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java]: fixed the checkstyle thing :-)

File: sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java [hunk 3]
```diff
@@ -3269,7 +3269,14 @@ private void appendDecoderDisplay(
             }
             else
             {
-                append(sb, indent, "builder.append('\\'').append(" + varDataName + "()).append('\\'');");
+                if (characterEncoding.contains("ASCII"))
+                {
+                    append(sb, indent, "builder.append('\\'');");
```
Comment (ariksher-nex) [inline sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java]: @mjpt777 do the single quotes around the string required?
care if i take those out? (both in ASCII and regular implementation)

Comment (mjpt777) [reply sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java]: Best to leave them.

**Pull request merged**